extensions/interactions/ImageClickInput/directives/ImageClickInput.js

```diff
@@ -44,12 +46,23 @@ oppia.directive('oppiaInteractiveImageClickInput', [
           $scope.highlightRegionsOnHover =
             ($attrs.highlightRegionsOnHoverWithValue === 'true');
           $scope.filepath = imageAndRegions.imagePath;
-          $scope.imageUrl = (
-            $scope.filepath ?
-              $sce.trustAsResourceUrl(
-                '/imagehandler/' +
-                ExplorationContextService.getExplorationId() +
-                '/' + encodeURIComponent($scope.filepath)) : null);
+          $scope.imageUrl = null;
+
+          if (ImagePreloaderService.inExplorationPlayer()) {
+            ImagePreloaderService.getImageUrl($scope.filepath
+            ).then(function(objectUrl) {
+              $scope.imageUrl = objectUrl;
+            });
```

Comment:
In this file just call `ImagePreloaderService.getImageUrl(` as before.

Reply:
Sorry didn't get you. You mean the original way. (not using the imagePreloaderService) ?

Reply:
We need to check whether we are playing an exploration or editing it. If we are in exploration player then the images are in the process of loading (ImagePreloaderService) but in case of editing the exploration, the images are not loaded because the service has not been called yet. We need to display the image for the preview, hence we need to check which mode is the exploration in. If in editing mode then use the AssetsBackendApiService for fetching the file directly.